### PR TITLE
Task/adq 151 im controller

### PR DESCRIPTION
**Pull request by bschofield-va**
Datamart immunization controller, transformer, and integration tests

### CODE REVIEW

**Comment by jblefkove-va** [comment]
image

File: data-query-tests/src/main/java/gov/va/api/health/dataquery/tests/SystemDefinitions.java [hunk 1]
```diff
@@ -106,7 +106,7 @@ private static TestIds localIds() {
         .diagnosticReport("1400096309217:L")
         .encounter("1200753214085")
         .diagnosticReports(diagnosticReports())
-        .immunization("1000000043979")
+        .immunization("12345678")
```
Comment (jblefkove-va) [inline data-query-tests/src/main/java/gov/va/api/health/dataquery/tests/SystemDefinitions.java]: I've got the same combination on my luggage!

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/immunization/DatamartImmunizationTransformer.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  private final DatamartImmunization datamart;
+
+  List<Annotation> note(Optional<String> note) {
+    return note.isPresent() ? List.of(Annotation.builder().text(note.get()).build()) : null;
+  }
+
+  List<Immunization.Reaction> reaction(Optional<DatamartReference> reaction) {
+    return reaction.isEmpty()
+        ? null
+        : List.of(Immunization.Reaction.builder().detail(asReference(reaction)).build());
```
Comment (jblefkove-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/immunization/DatamartImmunizationTransformer.java]: Is it safe to assume that if `reaction` is not empty, all the reference fields are populated and `asReference()` is guaranteed to return something?

File: data-query/src/main/resources/application.properties [hunk 3]
```diff
@@ -56,5 +56,6 @@ datamart.allergy-intolerance=false
 datamart.condition=false
 datamart.immunization=false
 datamart.diagnostic-report=true
+datamart.immunization=false
```
Comment (jblefkove-va) [inline data-query/src/main/resources/application.properties]: Repeated

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/immunization/DatamartImmunizationTransformer.java (hunk 2 above)
Comment (bschofield-va) [reply data-query/src/main/java/gov/va/api/health/dataquery/service/controller/immunization/DatamartImmunizationTransformer.java]: The _assumption_ is that if reaction is present then at least _display_ or, _type_ and _reference_ fields will be set.

Comment (bschofield-va) [reply data-query/src/main/java/gov/va/api/health/dataquery/service/controller/immunization/DatamartImmunizationTransformer.java]: I added some logic to help protect from an empty, but assigned reaction.

**Pull request merged**